13 update bmm examples

GidonFrischkorn opened the pull request:
This pull request contains the updated examples using the current `bmm` version. Prior to merging this into the `master` branch, we should consider creating a tag to allow users to access the scripts associated with the initial `bmm` release. In the long run this will likely be less relevant, but during the review process, I think it could be nice to be able to quickly switch back and forth between the different iterations of the Tutorial scripts.

In this pull request:
- All scripts are updated
- The figures were re-created using the new results to reflect potential changes due to informative default priors
- Some inferences using Bayes Factors were added for Example 2

venpopov commented:
Cool, thanks, I'll take a look now :)

venpopov commented on scripts/Parameter_Recovery/ParRecoverySimulation.R:
do we need this file? I started doing the parameter recovery for the mixture models, but haven't gotten to the IMM yet. In any case, this is unfinished, should we delete it?

GidonFrischkorn replied:
Yes.

venpopov commented on scripts/bmm_examples/Example1_Zhang&Luck2008.R:
```diff
@@ -69,74 +70,81 @@ head(data_ZL2008)
+  
+  backend = "cmdstanr",
+  refresh = 100,
+  file = here("output/fit_E1_ZL2008_bmm")
```
when I run this I get an error "Error: Object loaded via 'file' is not of class 'bmmfit'."

I think you might have ran this with an older version of bmm before we implemented the file argument - before we did that, the file argument was passed to brms instead, and the saved object was a brmsfit instead of a bmmfit object. Can you rerun all scripts with the newest bmm version and check if the files saved are bmmfit objects?

GidonFrischkorn replied:
Weird, when I load the file on my machine then I do not get the error message. I checked and the version that I used to estimate the models was `v0.5.1` and for me the file is also loaded as a `bmmfit` object.

venpopov replied:
Maybe the osf files are old? Or mine are? I'll try deleting all of them and re-download to see if my files are out dated and I'll let you know

venpopov replied:
I downloaded the files from osf, and the problem persists. When I fit the model myself it saves the correct bmmfit object. I suspect you might have uploaded old files? If the fits on your local machine are bmmfits, can you reupload them to OSF?

venpopov commented on scripts/bmm_examples/Example1_Zhang&Luck2008.R:
```diff
@@ -237,13 +245,25 @@ ggsave(
+plot(hypothesis(ZL_fit,hyp_ZL2008))
+
+# 5) Extract brms info from bmm object -----------------
+ZL_mockfit <- bmm(
```
why do we need a mockfit? the formula should be available from the bmmfit object

venpopov commented on scripts/bmm_examples/Example2_Loaiza&Souza2018.R:
```diff
@@ -74,86 +75,116 @@ levels(wmData$ageGroup) <- c("Young","Old")
 levels(wmData$cueCond) <- c("No","Retro")
 
+# specfiy contrasts
+contrasts(wmData$ageGroup) <- bayestestR::contr.equalprior(2)
```
I don't have this package installed so it gave me an error "Error in loadNamespace(x) : there is no package called ‘bayestestR’". Perhaps include it in the p_load function?

venpopov replied:
I'm also curious what this does as I haven't used it before, so maybe you could explain it to me (or is it in the paper already?) :)

GidonFrischkorn replied:
I have now included the package in the `p_load` statement. These priors are designed so that each condition mean has the same prior width after effects have been added up to the Intercept. For designs with factors that only have two-levels, like in the example, this is not super critical and you could also use `contr.sum` but as soon as a factor has 3 or more levels these contrasts are quite useful to have an equal prior width for all condition means: https://easystats.github.io/bayestestR/reference/contr.equalprior.html

venpopov commented on scripts/bmm_examples/Example2_Loaiza&Souza2018.R:
```diff
@@ -74,86 +75,116 @@ levels(wmData$ageGroup) <- c("Young","Old")
+# check the default priors
+default_prior(LS_formula, data = wmData, model = LS_model)
+
+prior <- 
```
something is missing here

GidonFrischkorn replied:
Deleted. This was left over from the bug in the `default_prior` function when I specified user priors to fit the model. I refit the model with the now correctly passed `default_priors`

venpopov commented on scripts/bmm_examples/Example2_Loaiza&Souza2018.R:
```diff
@@ -74,86 +75,116 @@ levels(wmData$ageGroup) <- c("Young","Old")
+                 max_treedepth = max_treedepth),
+  
+  # save the results
+  file = here("output","fit_E2_LS2018_bmm")
```
ok this one works! so it must be just a problem with the other script using an older bmm version?

venpopov commented on scripts/bmm_examples/Example3_Bays2009.R:
```diff
@@ -38,96 +38,98 @@ if (nChains >  parallel::detectCores()) {
+                                    max_treedepth = max_treedepth),
+                     
+                     # save results to file
+                     file = filename)
```
"Error: Object loaded via 'file' is not of class 'bmmfit'." - see my comment on Example 1

venpopov commented on scripts/bmm_examples/Example3_Bays2009.R:
```diff
@@ -38,96 +38,98 @@ if (nChains >  parallel::detectCores()) {
+pg <- exp(0) / (exp(thetat) + exp(thetant) + exp(0))
 
 # print parameter estimates
 kappa
```
won't let me comment on the clean_plot function because it's not changed code. Since you define that in an external function file and load, maybe get rid of the clean_plot definition here?

GidonFrischkorn replied:
Yes, deleted the function here.

venpopov commented on scripts/bmm_examples/Example4_unpublished_monotonic_constraints.R:
```diff
@@ -9,57 +9,88 @@
+                                      max_treedepth = max_treedepth),
+                       
+                       # save results to file
+                       file = filename)
```
"Error: Object loaded via 'file' is not of class 'bmmfit'." - see my comment on Example 1

venpopov commented on scripts/bmm_examples/Example4_unpublished_monotonic_constraints.R:
```diff
@@ -9,57 +9,88 @@
+                       # save results to file
+                       file = filename)
+
+fit_popov <- readRDS(here("output","fit_popov_unpublished_noconstraints_2p.rds"))
```
we wouldn't need this if you generate the fits with the latest version of bmm

venpopov commented on scripts/LoadResultsFiles.R:
need to remove the `rm(list=ls())` from this file - this file is sourced in the scripts after other files (e.g. the clean_plot()` script, and this causes problems in "Error in clean_plot() : could not find function "clean_plot"

venpopov commented on scripts/bmm_examples/Example5_Oberauer2017_3pMM.R:
The description at the top of the file states "#' This is the tutorial script for setting up the Interference Measurement Model (IMMabc)", but this is for the 3p mixture model. I also noticed some other inconsistent descriptions. It would be nice to update those to reflect the content of the files

venpopov commented on scripts/bmm_examples/Example5_Oberauer2017_3pMM.R:
```diff
@@ -41,67 +42,61 @@ adapt_delta <- .95
+                 max_treedepth = max_treedepth),
+  
+  # save results in file
+  file = filename
```
same issue as with the other files

venpopov commented on scripts/bmm_examples/Example5_Oberauer2017_IMMabc.R:
```diff
@@ -41,72 +42,59 @@ adapt_delta <- .95
+                 max_treedepth = max_treedepth),
+  
+  # save results to file
+  file = filename_IMMabc
```
same issue as with the other scripts

venpopov commented on scripts/bmm_examples/Example5_Oberauer2017_IMMbsc.R:
the plot for the s parameter has a y axis "General activation (a)"

venpopov reviewed (changes requested):
Overall this looks great! Thanks for updating everything. I left a few comments that need to be addressed, but otherwise it super nice to have this cleaned up and run with the new syntax :)

GidonFrischkorn commented:
I have now committed an updated version of the files addressing most of your comments. We should definately look into the issue with loading the files that I saved. On my machine I get no error messages and all the loaded objects have the class `bmmfit`. Maybe we need to have a look if the `file` option interacts with some other environment info that breaks portability between different setups.

venpopov commented:
> I have now committed an updated version of the files addressing most of your comments. We should definately look into the issue with loading the files that I saved. On my machine I get no error messages and all the loaded objects have the class `bmmfit`. Maybe we need to have a look if the `file` option interacts with some other environment info that breaks portability between different setups.

there shouldn't be a problem with the file option - both you and I when we run the models fresh we save the correct object. I suspect you might have uploaded some old fits to osf. Can you reupload your current fit objects after you confirm they are all bmmfits?

GidonFrischkorn commented:
Just as a short update. I will run the examples over the weekend with `v0.5.3` and then upload the new results files to OSF. I will sent you a notification once thats done, so that you can then check if all the files load correctly for you too.

GidonFrischkorn commented:
I now updated the `.rds` files on OSF and ran all models with the newest `bmm` version. Now, with downloading the new saved `bmmfits` the error when loading them should no longer appear. I also addressed your other comments.

venpopov commented:
Looks good and I can confirm all the bmm files now work for me as well.
